hicommonwealth/commonwealth
# 4680 - Typeahead Dropdown

This PR closes issue #4680. The card requires the creation of a typeahead select list, i.e. a select list that allows users to filter items in the list by their text input. Note that this select list also allows users to use it normally, i.e. scroll down and click on the option they want. Users also have the ability to click on an 'x' to clear their input. 

As of now, the only place in the app this list will appear is the related communities page (described in issue #4679), but this component is general use. It simply requires the options you wish to display, placeholder text (displayed in the input when no option is selected), and a default value (one of the options).

## Link to Issue
Closes: #4680

## Description of Changes
Creates a select list that does the following: 
  - allows user to select a list item by clicking it
  - allows user to select a list item by typing it in and hitting 'Enter' 
  - only allows a user to select one item
  - presents a default value to the user
  - displays 'No matches found' when their input does not match any list items

## Test Plan
- navigate to the demo site's [components page](https://commonwealth-demo.herokuapp.com/components)
- go to the 'Typeahead Dropdown' section and confirm that it behaves as expected (see list in previous section)

## Other Considerations
- required to complete #4679
